go-controller/pkg/util/kube.go

```diff
@@ -276,14 +280,34 @@ func UseEndpointSlices(kubeClient kubernetes.Interface) bool {
 }
 
 type LbEndpoints struct {
-	IPs  []string
+	EPs  []discovery.Endpoint
 	Port int32
 }
 
+func (e LbEndpoints) IPs() []string {
+	ipsSet := sets.NewString()
+	for _, endpoint := range e.EPs {
+		for _, ip := range endpoint.Addresses {
+			ipsSet.Insert(ip)
+		}
+	}
+	return ipsSet.List()
+}
+
+func (e LbEndpoints) LocalToNode(grName string) LbEndpoints {
+	res := LbEndpoints{Port: e.Port, EPs: make([]discovery.Endpoint, 0)}
+
+	for _, ep := range e.EPs {
+		if GetWorkerFromGatewayRouter(grName) == ep.Topology["kubernetes.io/hostname"] {
```

Comment:
I incidentally learned from @aojea that topology is going to be deprecated, and it's better to use the NodeName field https://github.com/kubernetes/kubernetes/commit/9613f79adde3e0080fe25576cb37a74f25262848

Reply:
Good Call, Thanks!